akka-http-core/src/test/scala/akka/http/impl/engine/client/LowLevelOutgoingConnectionSpec.scala

```diff
@@ -869,6 +869,28 @@ class LowLevelOutgoingConnectionSpec extends AkkaSpec("akka.loggers = []\n akka.
       netInSub.sendComplete()
       responses.expectComplete()
     }
+
+    "receive Content-Length for HEAD requests" in new TestSetup {
+      requestsSub.sendNext(HttpRequest(method = HttpMethods.HEAD))
+      expectWireData(
+        """HEAD / HTTP/1.1
+          |Host: example.com
+          |User-Agent: akka-http/test
+          |
+          |""")
+      sendWireData(
+        """HTTP/1.1 200 OK
+          |Content-Length: 6
+          |
+          |abcdef
```

Comment:
The response shouldn't have a body for a HEAD request but maybe add a content type instead.